subprojects/docs/src/docs/userguide/running-builds/introduction/build_file_basics.adoc

```diff
@@ -0,0 +1,189 @@
+plugins {
+    id 'org.jetbrains.kotlin.jvm' version '1.9.0'                       // <1>
+}
+
+repositories {
+    mavenCentral()                                                      // <2>
+}
+
+dependencies {
+    testImplementation 'org.jetbrains.kotlin:kotlin-test-junit5'        // <3>
+    testImplementation 'org.junit.jupiter:junit-jupiter-engine:5.9.3'
+    testRuntimeOnly 'org.junit.platform:junit-platform-launcher'
+    implementation 'com.google.guava:guava:32.1.1-jre'
+}
+
+application {
+    mainClass = 'main'                                                  // <4>
+}
+
+tasks.named('test') {
+    useJUnitPlatform()                                                  // <5>
+}
+----
+<1> Add plugins.
+<2> Add dependency location.
+<3> Add dependencies.
+<4> Use convention properties.
+<5> Register or configure tasks.
+=====
+====
+
+=== 1. Add plugins
+Plugins are bundled tasks used to extend Gradle.
+
+Adding a plugin to a build is called _applying_ a plugin.
+
+[source]
+----
+plugins {
+    id("org.jetbrains.kotlin.jvm") version "1.9.0"
+}
+----
+
+The Kotlin gradle plugin, version 1.9.0, has been applied to the project so that source code written in Kotlin can be used.
+
+=== 2. Add dependency location
+A project generally has several dependencies it needs to do its work.
+Many of the dependencies will be used by the source code.
+
+In the example, the link:https://mvnrepository.com/[Maven Central^] repository has been added as the location for all the binaries that Gradle will need to download to build the project.
+
+[source]
+----
+repositories {
+    mavenCentral()
+}
+----
+
+In this example, the Kotlin JVM Plugin, `org.jetbrains.kotlin.jvm`, will be downloaded by Gradle from the link:https://mvnrepository.com/artifact/org.jetbrains.kotlin.jvm/org.jetbrains.kotlin.jvm.gradle.plugin[Maven Central Repository].
+
+=== 3. Add dependencies
+Dependencies are grouped in configurations and can be retrieved and uploaded from repositories.
+
+In the example, a list of dependencies based on configurations (how/when/where Gradle will use it) is added, including the `junit` testing framework, which is used to test the Java source code.
+
+[source]
+----
+dependencies {
+    testImplementation("org.jetbrains.kotlin:kotlin-test-junit5")
+    testImplementation("org.junit.jupiter:junit-jupiter-engine:5.9.3")
+    testRuntimeOnly("org.junit.platform:junit-platform-launcher")
+    implementation("com.google.guava:guava:32.1.1-jre")
+}
+----
+
+In this example, the source code in the project will import the Google `guava` library which will also be retrieved from the link:https://mvnrepository.com/artifact/com.google.guava/guava[Maven Central Repository] by Gradle.
+
+=== 4. Use convention properties
+A plugin adds tasks to a project.
+It also adds properties and methods to a project.
+
+In the example, the `application` plugin has a method that can declare the main class of a Java application, which is required to compile and run the code.
```

Comment:
```suggestion
A plugin adds tasks to a project.
It can also add properties to a project via an extension.

In the example, the `application` plugin adds an extension with a property that controls the main class of a Java application, which is required to compile and run the code.
```

I think we should probably use terminology that is conventional in Gradle. Particularly, it seems odd to say `the application plugin has a method`. The plugin itself does not have a method, but rather registers an extension on the project that does